plugins/azure/keyvaults/checkEncryptionKeysExpiry.js

```diff
@@ -0,0 +1,69 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Key Expiration Enabled',
+    category: 'Key Vaults',
+    domain: 'Identity and Access Management',
+    description: 'Ensure that all Keys in Azure Key Vault have an expiry time set.',
+    more_info: 'Setting an expiry time on all keys forces key rotation and removes unused and forgotten keys from being used.',
+    recommended_action: 'Ensure each Key Vault has an expiry time set that provides for sufficient rotation.',
+    link: 'https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates',
+    apis: ['vaults:list', 'vaults:getKeys'],
+
```

Comment:
Let's take user input for how many days before key expires, we should give a fail result as a setting. Default can be 30. 
key_vault_key_expiry_fail: {
            name: 'Key Vault Key Expiry Fail',
            description: 'Return a failing result when key expiration date is within this number of days in the future',
            regex: '^[1-9]{1}[0-9]{0,3}$',
            default: '30'
        }